src/shared/components/case-editor/services/wizard-page-field-to-case-field.mapper.ts

```diff
@@ -0,0 +1,88 @@
+import { WizardPageField } from '../domain';
+import { CaseField } from '../../../domain';
+import { ComplexFieldOverride } from '../domain/wizard-page-field-complex-override.model';
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class WizardPageFieldToCaseFieldMapper {
+
+  mapAll(wizardPageFields: WizardPageField[], caseFields: CaseField[]): CaseField[] {
+    return wizardPageFields.map(wizardField => {
+      return this.map(wizardField, caseFields);
+    });
+  }
+
+  private map(wizardPageField: WizardPageField, caseFields: CaseField[]): CaseField {
+
+    let caseField: CaseField = caseFields.find(e => e.id === wizardPageField.case_field_id);
+    caseField.id = wizardPageField.case_field_id;
+    caseField.wizardProps = wizardPageField;
+    caseField.display_context = wizardPageField.display_context;
+    caseField.order = wizardPageField.order;
+
+    if (wizardPageField.complex_field_overrides && wizardPageField.complex_field_overrides.length > 0) {
+      wizardPageField.complex_field_overrides.forEach((override: ComplexFieldOverride) => {
+        this.processComplexFieldOverride(override, caseField, caseFields);
+      });
+    }
+
+    return caseField;
+  }
+
+  private processComplexFieldOverride(override: ComplexFieldOverride, case_field: CaseField, caseFields: CaseField[]) {
+    const caseFieldIds = override.complex_field_element_id.split('.');
+    let case_field_leaf: CaseField;
```

Comment:
can we use camel case please? `case_field_leaf` to `caseFieldLeaf` or, if you will, also `overriddenField` or `overriddenSubfield` would be more self explanatory?

Reply:
👍